mobile-ui/src/main/java/org/sagebionetworks/research/mobile_ui/perform_task/PerformTaskFragment.java

```diff
@@ -170,9 +176,10 @@ public void onCreate(Bundle savedInstanceState) {
         }
 
         TaskResult taskResult = new TaskResultBase("taskID", Instant.now(), UUID.randomUUID());
+        sharedPrefsArgs = getSharedPrefsArgs();
         performTaskViewModel = ViewModelProviders
                 .of(this, taskViewModelFactory.create(taskView, taskRunParcelableUuid.getUuid(),
-                        this.getLastRun()))
+                        sharedPrefsArgs))
```

Comment:
I think it'd make more sense for the ViewModel to be performing logic

Reply:
As in the view model should be getting the arguments from SharedPrefs?

Reply:
Yes